src/main/java/overlap_use_case/TimetableModelUnpacker.java

```diff
@@ -0,0 +1,95 @@
+package overlap_use_case;
+
+import entities.*;
+import retrieve_timetable_use_case.*;
+
+import java.util.ArrayList;
+
+/** A class responsible for converting between Timetable Model classes (that is, including its internal
+ * model classes) back into their entity forms. "Unpacks the model class" into the entity.
+ */
+public class TimetableModelUnpacker {
+
+    public static Session unpackSessionModel(SessionModel sessionModel){
+
+        Session session = new Session(sessionModel.getSessionType());
+
+        for (CourseModel courseModel : sessionModel.getCourses().values()){
+            session.addCourse(unpackCourseModelAsCalendarCourse(courseModel));
+        }
+
+        return session;
+    }
+
+    public static Timetable unpackTimeTable(TimetableModel timetable){
+        ArrayList<TimetableCourse> courses = new ArrayList<>();
+        for (CourseModel course : timetable.getCourses() ){
+            courses.add(unpackCourseModelasTimetableCourse(course));
+        }
+
+        return new Timetable(courses, timetable.getSessionType());
+    }
+
+    // Since this class is specifically for unpacking Timetables, we make a TimetableCourse.
+    public static TimetableCourse unpackCourseModelasTimetableCourse(CourseModel course){
+
+        ArrayList<Section> sections = new ArrayList<>();
+        for (SectionModel sectionModel : course.getSections()){
+            sections.add(unpackSectionModel(sectionModel));
+        }
+
+        try {
+            return new TimetableCourse(course.getTitle(), sections, course.getCourseSession(),
+                    course.getCourseCode(), course.getBreadth());
+        } catch (InvalidSectionsException e){
+            throw new IllegalArgumentException("You tried to unpack a CourseModel that has invalid sections!");
+        }
+    }
+
+    // FIXME: I think this might be a code smell. Two nearly identical methods with one slightly changed constructor call.
+    public static CalendarCourse unpackCourseModelAsCalendarCourse(CourseModel course){
+
+        ArrayList<Section> sections = new ArrayList<>();
+        for (SectionModel sectionModel : course.getSections()){
+            sections.add(unpackSectionModel(sectionModel));
+        }
+            return new CalendarCourse(course.getTitle(), sections, course.getCourseSession(),
+                    course.getCourseCode(), course.getBreadth());
+
+    }
+
+    public static Section unpackSectionModel(SectionModel section){
+        ArrayList<Block> blocks = new ArrayList<>();
+        for(BlockModel blockModel : section.getBlocks()){
+            blocks.add(unpackBlockModel(blockModel));
+        }
+        return new Section(section.getCode(), section.getInstructor(), blocks);
+    }
+
+    public static Block unpackBlockModel(BlockModel block){
+        String blockDay = switch (block.getDay()) {
```

Comment:
It looks like this particular set of switch case statements is breaking the autograder. Is there any way for you to refactor it into something the autograder will accept without raising an IntelliJ warning?

Reply:
Tried it! Strange since this is just syntax sugar (it seems like a valid expression, at least in the version of Java we're using!).
Hopefully this more conventional version holds :D.